### ISSUE TITLE
JWT and KONG with custom authorizations

### ISSUE THREAD

**Issue by ksachdev1** [Kong/kong#2199]
### Summary

SUMMARY_GOES_HERE
I have gone through Kong JWT plugin and the quick start tutorials. I am unable to figure out, if Kong can let me add custom user roles in the payload, that I can relay to my microservices to do authorizations

### Steps To Reproduce
I went through this tutorial on KONG https://getkong.org/plugins/jwt/

I have an understanding of JWT and authorization concepts. I have prototyped JWT with Spring Boot where I could put my own key value like this {"authorizations":"role_admin, role_user"}.

It is easy to do that in Spring Boot but I am not able to find information on how to do this with KONG. Anyone has any info about it?

Basically, I want to add my own authorization list and then decide if the api that is being accessed by the consumer, if the consumer has any or all of those authorizations

### Additional Details & Logs

- Kong version (`$ kong version`)
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
- Kong configuration (registered APIs/Plugins & configuration file)
- Operating System

**Comment by jeanpralo** [comment]
You can use the acl plugin for that: https://getkong.org/plugins/acl/


I am currently working on a plugin that reads the role field inside a JWT since the default plugin does not do it.

**Comment by ksachdev1** [comment]
@jeanpralo I took a quick look at that link. Few questions, pardon me if I have missed in my quick reading. 
1. While creating a use I would add them to a group, like say group_employee, group_admin and then use ACL?
2. If (1) above is correct, are there ways to say more granual things like if for an api i say "group1, group2", the api can be executed if user belongs to both, or one or any in the api list? Are these configs possible?

**Comment by jeanpralo** [comment]
I think this discussion would be better off here: https://groups.google.com/forum/#!forum/konglayer


@ksachdev1  I am unsure I am not using that plugin, I would say refer to the doc.

**Comment by ksachdev1** [comment]
@jeanpralo , i posted it there before posting here, but did not get any response.
https://groups.google.com/forum/#!topic/konglayer/4S6RqrLhJrc

**Comment by omri-shaiko** [comment]
Hi @ksachdev1 
I'm facing the same issues as you. have you find any solution for it? did you wrote a plugin for it yourself?

Thanks

**Comment by dkavraal** [comment]
Same here.

**Comment by ksachdev1** [comment]
@omri-shaiko and @dkavraal , We built authorization logic at the microservice layer. We use kong for token validation only now.

**Comment by hishamhm** [comment]
Hi there, a better venue for feature suggestions is now the [Kong Nation](http://discuss.konghq.com) forum! I'm closing this one for now as we are keeping the Github issue tracker focused on bug reports. Feel free to start a topic there in the "Feature Suggestions" area so this doesn't get mixed alongside other issues. Thank you!

**Comment by mssaisandeep** [comment]
> You can use the acl plugin for that: https://getkong.org/plugins/acl/
> 
> I am currently working on a plugin that reads the role field inside a JWT since the default plugin does not do it.

Hey, did you got any plugin that reads permissions/roles from JWT Token and makes decisions on top of that? Please help me. I am also looking for that.

**Comment by mssaisandeep** [comment]
@jeanpralo Hey could you please help me with the plugin that you mentioned previously - Reading claims from token and performing authorization decisions based on that. If there are any pointers that you can point me to, that would be great. Thanks.

**Comment by biqbalgis** [comment]
> @omri-shaiko and @dkavraal , We built authorization logic at the microservice layer. We use kong for token validation only now.

Could you please post some details? I am interested in the same

**Comment by spmsupun** [comment]
Anyone got solution for this ?
